src/coreclr/jit/lsraloongarch64.cpp

```diff
@@ -0,0 +1,1373 @@
+                buildInternalIntRegisterDefForNode(indirTree);
+            }
+        }
+        else if (addr->OperGet() == GT_CLS_VAR_ADDR)
+        {
+            // Reserve int to load constant from memory (IF_LARGELDC)
+            buildInternalIntRegisterDefForNode(indirTree);
+        }
+    }
+
+#ifdef FEATURE_SIMD
+    if (indirTree->TypeGet() == TYP_SIMD12)
+    {
+        // If indirTree is of TYP_SIMD12, addr is not contained. See comment in LowerIndir().
+        assert(!addr->isContained());
+
+        // Vector3 is read/written as two reads/writes: 8 byte and 4 byte.
+        // To assemble the vector properly we would need an additional int register
+        buildInternalIntRegisterDefForNode(indirTree);
+    }
+#endif // FEATURE_SIMD
+
+    int srcCount = BuildIndirUses(indirTree);
+    buildInternalRegisterUses();
+
+    if (!indirTree->OperIs(GT_STOREIND, GT_NULLCHECK))
+    {
+        BuildDef(indirTree);
+    }
+    return srcCount;
+}
+
+//------------------------------------------------------------------------
+// BuildCall: Set the NodeInfo for a call.
+//
+// Arguments:
+//    call - The call node of interest
+//
+// Return Value:
+//    The number of sources consumed by this node.
+//
+int LinearScan::BuildCall(GenTreeCall* call)
+{
+    bool                  hasMultiRegRetVal = false;
+    const ReturnTypeDesc* retTypeDesc       = nullptr;
+    regMaskTP             dstCandidates     = RBM_NONE;
+
+    int srcCount = 0;
+    int dstCount = 0;
+    if (call->TypeGet() != TYP_VOID)
+    {
+        hasMultiRegRetVal = call->HasMultiRegRetVal();
+        if (hasMultiRegRetVal)
+        {
+            // dst count = number of registers in which the value is returned by call
+            retTypeDesc = call->GetReturnTypeDesc();
+            dstCount    = retTypeDesc->GetReturnRegCount();
+        }
+        else
+        {
+            dstCount = 1;
+        }
+    }
+
+    GenTree*  ctrlExpr           = call->gtControlExpr;
+    regMaskTP ctrlExprCandidates = RBM_NONE;
+    if (call->gtCallType == CT_INDIRECT)
+    {
+        // either gtControlExpr != null or gtCallAddr != null.
+        // Both cannot be non-null at the same time.
+        assert(ctrlExpr == nullptr);
+        assert(call->gtCallAddr != nullptr);
+        ctrlExpr = call->gtCallAddr;
+    }
+
+    // set reg requirements on call target represented as control sequence.
+    if (ctrlExpr != nullptr)
+    {
+        // we should never see a gtControlExpr whose type is void.
+        assert(ctrlExpr->TypeGet() != TYP_VOID);
+
+        // In case of fast tail implemented as jmp, make sure that gtControlExpr is
+        // computed into a register.
+        if (call->IsFastTailCall())
+        {
+            // Fast tail call - make sure that call target is always computed in volatile registers
+            // that will not be overridden by epilog sequence.
+            ctrlExprCandidates = allRegs(TYP_INT) & RBM_INT_CALLEE_TRASH;
```

Comment:
There was a change recently around here. Double check if it is needed for loonarch64.
https://github.com/dotnet/runtime/pull/66282/files#diff-41f955382047ac3aede3f2a10d224736e9267c50fb97c881db4db8cb52574e91R181-R185

Reply:
OK, Thanks very much.
I will check it.